### ISSUE TITLE
Symlink cycles in project cause server to use 100% CPU and keep running after quit

### ISSUE THREAD

**Issue by maxbrunsfeld** [atom/ide-typescript#95]
When working on JavaScript projects that have cyclical symlink paths, my CPU gets maxed out by one or more `javascript-typescript-langserver/lib/language-server-stdio` processes. These processes use 100% CPU and keep running even after I reload an Atom window or quit Atom.

![maxed-out-cpu](https://user-images.githubusercontent.com/326587/36869816-f228eef2-1d51-11e8-8f77-d414e2362685.png)

I have attached a tarball of a minimal directory structure that produces the problem:

```
issue-repro $ tree
.
├── test1
│   ├── index.js
│   ├── node_modules
│   │   ├── test2 -> ../../test2
│   │   └── test3 -> ../../test3
│   └── package.json
├── test2
│   ├── index.js
│   ├── node_modules
│   │   ├── test1 -> ../../test1
│   │   └── test3 -> ../../test3
│   └── package.json
└── test3
    ├── index.js
    ├── node_modules
    │   ├── test1 -> ../../test1
    │   └── test2 -> ../../test2
    └── package.json

```

[issue-repro.tar.zip](https://github.com/atom/ide-typescript/files/1772793/issue-repro.tar.zip)

This is probably initially caused by a bug in [the language server itself](https://github.com/sourcegraph/javascript-typescript-langserver), but it also seems like this package or `atom-languageclient` should not allow their child processes to continue running like this after shutdown.

/cc @damieng

**Comment by erkie** [comment]
This issue has been closed on the language server repo: https://github.com/sourcegraph/javascript-typescript-langserver/issues/453 so I doubt it will be fixed. @maxbrunsfeld in the linked issue the maintainer assumed cyclical imports was bad and thus not a problem on their end. How did you end up  in your situation? Maybe we can make a good case for this being something to actually look into.

**Comment by damieng** [comment]
Likely fixed with the new ide-typescript 0.8.0 update.
